api/alert_rules.go

```diff
@@ -0,0 +1,201 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// AlertRulesService is the service that interacts with
+// the AlertRules schema from the Lacework APIv2 Server
+type AlertRulesService struct {
+	client *Client
+}
+
+type alertRuleType string
+
+type alertRuleSeverity int
+
+type AlertRuleSeverities []alertRuleSeverity
+
+// Alert Rule type is an enum in api v2, with only 1 supported value "Event"
+const AlertRuleEventType alertRuleType = "Event"
+
+// String returns the string representation of a Alert Rule integration type
+func (i alertRuleType) String() string {
+	return string(i)
+}
+
+func (sevs AlertRuleSeverities) toInt() []int {
+	var res []int
+	for _, i := range sevs {
+		res = append(res, int(i))
+	}
+	return res
+}
+
+const (
+	AlertRuleSeverityCritical alertRuleSeverity = iota + 1
+	AlertRuleSeverityHigh
+	AlertRuleSeverityMedium
+	AlertRuleSeverityLow
+	AlertRuleSeverityInfo
+)
+
+// NewAlertRule returns an instance of the AlertRuleData struct
+//
+// Basic usage: Initialize a new AlertRuleData struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   alertRule := api.NewAlertRule(
+//		"Foo",
+//     api.AlertRule{
+//		Description: "My Alert Rule"
+//		Severities: api.AlertRuleSeverities{api.AlertRuleSeverityHigh,
+//		Channels: []string{"TECHALLY_000000000000AAAAAAAAAAAAAAAAAAAA"},
+//		ResourceGroups: []string{"TECHALLY_111111111111AAAAAAAAAAAAAAAAAAAA"}
+//       },
+//     },
+//   )
+//
+//   client.V2.AlertRules.Create(alertRule)
+//
+func NewAlertRule(name string, rule AlertRule) AlertRuleData {
+	return AlertRuleData{
+		Channels: rule.Channels,
+		Type:     AlertRuleEventType.String(),
+		Filter: AlertRuleFilter{
+			AlertRuleComputedData: AlertRuleComputedData{},
+			Name:                  name,
+			Enabled:               1,
+			Description:           rule.Description,
+			Severity:              rule.Severities.toInt(),
+			ResourceGroups:        rule.ResourceGroups,
+			EventCategories:       rule.EventCategories,
+		},
+	}
+}
+
+// List returns a list of Alert Rules
+func (svc *AlertRulesService) List() (response AlertRulesResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2AlertRules, nil, &response)
+	return
+}
+
+// Create creates a single Alert Rule
+func (svc *AlertRulesService) Create(rule AlertRuleData) (
+	response AlertRuleResponse,
+	err error,
+) {
+	err = svc.create(rule, &response)
```

Comment:
we created private functions when we had multiple types and every type was calling the underlying service function, this is a unique type so, having separate functions for `create()` `update()` etc. feels like a copy-paste that is not needed for this Alert Rule service.

Can we just do the API call directly here and delete the other private functions?
```suggestion
	err = svc.client.RequestEncoderDecoder("POST", apiV2AlertRules, &rule, &response)
```
Something like that 👆🏽